keyboards/abko/ak84bt/keymaps/default/keymap.c

```diff
@@ -0,0 +1,180 @@
+    XXXXXXX , RGB_TOG  , RGB_MOD , XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX ,   XXXXXXX , XXXXXXX , KC_BK_LIGHTNESS_UP , XXXXXXX , XXXXXXX ,
+    XXXXXXX , KC_WINLOCK_ONOFF, XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX , KC_BK_LIGHT_ONOFF, XXXXXXX , XXXXXXX , XXXXXXX  , MO(_FnLay), XXXXXXX , KC_BK_LIGHT_SPEED_DOWN , KC_BK_LIGHTNESS_DOWN , KC_BK_LIGHT_SPEED_UP , XXXXXXX
+    )    
+};
+
+void keyboard_post_init_user(void)
+{ 
+}
+
+void keyboard_pre_init_user(void)
+{
+}
+
+bool winlock_state = 0;
+bool process_record_user(uint16_t keycode, keyrecord_t *record)
+{    
+    // If console is enabled, it will print the matrix position and status of each key pressed
+#ifdef CONSOLE_ENABLE
+    uprintf("KL: kc: 0x%04X, col: %2u, row: %2u, pressed: %u, time: %5u, int: %u, count: %u\r\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed, record->event.time, record->tap.interrupted, record->tap.count);
+#endif    
+
+    switch (keycode)
+    {                             
+        case KC_WINLOCK_ONOFF:
+        {
+            if (record->event.pressed)
+            {
+                winlock_state ^= 1;
+            }
+
+            return false;
+        }
+
+        case KC_BK_LIGHT_ONOFF:
+        {
+#ifdef RGB_MATRIX_ENABLE    
+            if (record->event.pressed)
+            {        
+                rgb_matrix_toggle();
+            }
+#endif
+
+            return false;
+        }        
+
+        case KC_BK_LIGHTNESS_UP:
+        {
+#ifdef RGB_MATRIX_ENABLE            
+            if (record->event.pressed)
+            {        
+                rgb_matrix_increase_val();
+            }
+#endif            
+            return false;
+        }
+
+        case KC_BK_LIGHTNESS_DOWN:
+        {
+#ifdef RGB_MATRIX_ENABLE            
+            if (record->event.pressed)
+            {        
+                rgb_matrix_decrease_val();
+            }
+#endif            
+            return false;
+        }
+
+        case KC_BK_LIGHT_SPEED_UP:
+        {
+#ifdef RGB_MATRIX_ENABLE            
+            if (record->event.pressed)
+            {        
+                rgb_matrix_increase_speed();
+            }
+#endif            
+            return false;
+        }
+
+        case KC_BK_LIGHT_SPEED_DOWN:
+        {
+#ifdef RGB_MATRIX_ENABLE            
+            if (record->event.pressed)
+            {        
+                rgb_matrix_decrease_speed();
+            }
+#endif            
+            return false;
+        }
+        
+        case KC_RGB_SWITCH_EFFECT:
+        {
+#ifdef RGB_MATRIX_ENABLE            
+            if (record->event.pressed)
+            {        
+                rgb_matrix_step();
+            }
+#endif            
+            return false;
+        }
+
```

Comment:
All these custom keycodes are doing what the [core](https://docs.qmk.fm/#/feature_rgb_matrix?id=keycodes) ones already do. Those should be used instead. 
```suggestion
```